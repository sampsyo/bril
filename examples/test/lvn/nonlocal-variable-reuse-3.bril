# CMD: bril2json < {filename} | python3 ../../lvn.py -p | bril2txt
@main {
  # This should re-name within local scope only with `lvn`.
  a: int = const 3;
  b: int = id a;
  a: int = id b;
  print b;
  print a;
  a: int = const 4;
  print a;
  a: int = const 5;
  print a;
}
