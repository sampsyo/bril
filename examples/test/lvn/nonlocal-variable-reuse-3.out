@main {
  lvn.0: int = const 3;
  b: int = const 3;
  a: int = const 3;
  print lvn.0;
  print lvn.0;
  lvn.1: int = const 4;
  print lvn.1;
  a: int = const 5;
  print a;
}
